include/nbl/asset/ICPUComputePipeline.h

```diff
@@ -25,7 +25,7 @@ class ICPUComputePipeline : public ICPUPipeline<IPipeline<ICPUPipelineLayout>,1>
         {
             if (!params.layout)
                 return nullptr;
-            auto retval = new ICPUComputePipeline(core::smart_refctd_ptr<const ICPUPipelineLayout>(params.layout));
+            auto retval = new ICPUComputePipeline(core::smart_refctd_ptr<const ICPUPipelineLayout>(params.layout), EPBP_COMPUTE);
```

Comment:
why are you passing an enum into the constructor of a derived class?

the derived class should know its own type thanks to the vtable